### ISSUE TITLE
Zero-length file when downloading apktool_2.0.3.jar

### ISSUE THREAD

**Issue by chetbox** [iBotPeaches/Apktool#1176]
When I download https://bitbucket.org/iBotPeaches/apktool/downloads/apktool_2.0.3.jar the download succeeds but the file is zero-length.

This is currently breaking my automated build. All other downloads from BitBucket seem to be working correctly so it seems like `apktool_2.0.3.jar` may have been replaced with an empty file.

**Comment by chetbox** [comment]
Just seen the [latest commit](https://github.com/iBotPeaches/Apktool/commit/57a8eb39b8e4e9c034d295fdcef6be055481df3d) which acknowledges this problem. For those looking for an alternative download location please use http://connortumbleson.com/apktool/apktool_2.0.3.jar

**Comment by iBotPeaches** [comment]
Yeah, working with support at Bitbucket now. Appears things aren't working quite right, so I'll have to reupload the binaries to them soon.

Though GitHub now supports file uploads again, so it may be time to bring everything under the same platform again vs just using BitBucket for the downloads. (Since Googlecode shut down containing the downloads and GitHub didn't allow creation of custom downloads back then)

Will keep this report open till the binaries are solved.

**Comment by iBotPeaches** [comment]
Fixed binaries are on BitBucket. The solution from support was to delete the binaries and re-upload them. I also mirrored the 2.0.3 release to GitHub here - https://github.com/iBotPeaches/Apktool/releases/tag/2.0.3

I like that GitHub allows file uploads, but then it confuses the user by putting download buttons directly below the binary file itself. This will probably have users download the ZIP package thinking its a zip of the JAR, but in fact its a ZIP of the project which will need to be built and what not. I don't like the UI there at all. I also dislike that the download count is hidden behind the API.

I tend to use the download count to gauge releases.

TLDR - Old, but new binaries uploaded to Bitbucket. Crises averted. Change might be coming.

**Comment by chetbox** [comment]
Thanks for the quick turnaround. Much appreciated.

**Comment by chetbox** [comment]
And w.r.t the download buttons below the binary download button, IMHO it's fairly clear that you'll get a copy of the source code (and no build) given that the links have "Source code" next to them.

In general, I'm in favour of keeping everything in one place. When I first started using apktool I assumed that BitBucket was a third-party mirror and spent a while searching for the "official" releases which I'd already found!

**Comment by iBotPeaches** [comment]
Yeah its been messy. Used to be only Googlecode. Then Googlecode removed the ability to upload files and then completely shut down a ~year later.

At that time 2 years ago, GitHub had removed the ability to upload releases and the only options were self host or use Bitbucket. Today, our options and my personal server is a bit more equipped. I'll weigh the options, but might move towards GitHub as the primary location for binaries with my personal server as a mirror.

**Comment by iBotPeaches** [comment]
Post Mortem - https://github.com/iBotPeaches/Apktool/commit/93ab25887db150525ed70a89138ac514d6578f5f#commitcomment-16560635
